pkg/recipes/terraform/config/providers/aws.go

```diff
@@ -132,9 +143,18 @@ func (p *awsProvider) generateProviderConfigMap(credentials *credentials.AWSCred
 		config[awsRegionParam] = region
 	}
 
-	if credentials != nil && credentials.AccessKeyID != "" && credentials.SecretAccessKey != "" {
-		config[awsAccessKeyParam] = credentials.AccessKeyID
-		config[awsSecretKeyParam] = credentials.SecretAccessKey
+	if credentials != nil {
+		if credentials.Kind == ucp_datamodel.AWSAccessKeyCredentialKind {
+			if credentials.AccessKeyCredential != nil &&
+				credentials.AccessKeyCredential.AccessKeyID != "" && credentials.AccessKeyCredential.SecretAccessKey != "" {
+				config[awsAccessKeyParam] = credentials.AccessKeyCredential.AccessKeyID
+				config[awsSecretKeyParam] = credentials.AccessKeyCredential.SecretAccessKey
+			}
+		} else if credentials.Kind == ucp_datamodel.AWSIRSACredentialKind {
+			if credentials.IRSACredential != nil && credentials.IRSACredential.RoleARN != "" {
+				config[awsRoleARN] = credentials.IRSACredential.RoleARN
```

Comment:
is this how to enable irsa with terraform? can we link a terraform docs page here?

Reply:
removed terraform related changes ref. https://github.com/radius-project/radius/pull/7739#discussion_r1679825544